tycho-indexer/src/services/rpc.rs

```diff
@@ -718,6 +732,144 @@ where
+            .get_entry_points_tracing_params(filter, Some(&pagination_params))
+            .await
+            .map_err(|err| {
+                error!(error = %err, "Error while getting entry points with tracing params.");
+                err
+            })?;
+
+        trace!(
+            entry_points_tracing_params = ?entry_points_tracing_params_data,
+            "Retrieved entry points with tracing params from database."
+        );
+
+        // Flatten the ID lists, throwing away component ids, to avoid making duplicate db calls
+        // when getting traced entry points.
+        let entry_point_ids: HashSet<EntryPointId> = entry_points_tracing_params_data
+            .entity
+            .values()
+            .flat_map(|entry_points_with_tracing_params| {
+                entry_points_with_tracing_params
+                    .iter()
+                    .map(|entry_point| {
+                        entry_point
+                            .entry_point
+                            .external_id
+                            .clone()
+                    })
+            })
+            .collect();
+
+        let traced_entry_points = self
+            .db_gateway
+            .get_traced_entry_points(&entry_point_ids)
+            .await
+            .map_err(|err| {
+                error!(error = %err, "Error while getting traced entry points.");
+                err
+            })?;
+
+        trace!(
+            traced_entry_points = ?traced_entry_points,
+            "Retrieved traced entry points entry points from database."
+        );
+
+        let mut traced_entry_points_by_component = HashMap::new();
+
+        for (component_id, entry_points_set) in entry_points_tracing_params_data.entity {
+            let mut pairs = Vec::new();
+
+            for entry_point_with_params in entry_points_set {
+                let entry_point_id = entry_point_with_params
+                    .entry_point
+                    .external_id
+                    .as_str();
+                let tracing_param = &entry_point_with_params.params;
+                let results_by_param = traced_entry_points
+                    .get(entry_point_id)
+                    .ok_or_else(|| {
+                        RpcError::Storage(StorageError::NotFound(
+                            "TracedEntryPoint".to_string(),
+                            format!("{entry_point_id:?}"),
+                        ))
+                    })?;
+
+                let tracing_result = results_by_param
+                    .get(tracing_param)
+                    .ok_or_else(|| {
+                        RpcError::Storage(StorageError::NotFound(
+                            "TracedEntryPoint".to_string(),
+                            format!("{entry_point_id:?} with params {tracing_param:?}"),
+                        ))
+                    })?;
+
+                pairs.push((entry_point_with_params.clone().into(), tracing_result.clone().into()));
```

Comment:
We should be able to avoid at least 1 clone here:
```suggestion
                pairs.push((entry_point_with_params.into(), tracing_result.clone().into()));
```